templates/fine-tune-llm_v2/cookbooks/continue_from_checkpoint/README.md

```diff
@@ -0,0 +1,57 @@
+# Continue fine-tuning from a previous checkpointing
+
+**⏱️ Time to complete**: 40 minutes
+
+This document assumes that you have familiarized yourself with the main fine-tuning guide of this template.
+In this folder of the template, we showcase how a checkpoint that was created earlier can be used to start a training from.
+We case use this, for example, if we think that starting from a given checkpoint will give us a performance advantage.
+
+There are two types of checkpoints to considere here: Full-parameter checkpoints, and LoRA-adapter checkpoints.
+For starters, we advise against combining the two (by training a LoRA adapter ontop of a full-parameter checkpoint), because serving the resulting LoRA adapter will require the full-parameter checkpoint.
+
+## How to fine-tune from a previous checkpointing
+
+
+```python
+# [Optional] You can set the WandB API key to track model performance
+# !export WANDB_API_KEY={YOUR_WANDB_API_KEY}
+
+# Continue LoRA fine-tuning on the GSM8k dataset with Llama 3 8B
+!llmforge dev finetune llama-3-8b.yaml
+```
+
+## What we are fine-tuning on
+
+Running the above command will fine-tune on the [GSM8k dataset](https://huggingface.co/datasets/gsm8k). 
+In this example, we split the dataset into two halfs, each consisting of approximately 4.000 samples.
+The provided initial checkpoint has been trained on the first half and is already good at solving GSM8k. By running the above command, you continue fine-tuning from the provided checkpoint with the second half.
+
+## Inference and evaluation
+
+In order to evaluate the produced checkpoint, you need to serve it. You can do this with the LLM inference template.
+On the cluster that you are using to serve the fine-tuned model, you can use the provided script `evaluate_gsm8k.py` to evaluate.
+Don't forget to change the `ADAPTER_NAME` variable to fit the name of your model.
+
+# How to use this for your own purpose
+
+The training and evaluation loss of the second, the continued, fine-tuning are lower than what we saw in the first run.
+For example, the checkpoint that you find in the llama-3-8b.yaml has an evaluation loss of 0.8886.
+After continued fine-tuning, we reach a checkpoint with an evaluation loss of 0.8668.
+Such loss values depend greatly on the task at hand - a difference of 0.0218 may be a big improvement on some tasks and a minor improvement on others.
+
+We advise to monitor training loss and evaluation loss of fine-tunes to find out if you are improving through the continued fine-tuning.
```

Comment:
I would write this part differently. 

1. I would talk about the diffs in terms of loss value via showing the diffs in the loss curve and zooming / annotating the parts that makes it super clear how continued ft helped.
2. I would not get into the details of loss values themselves, just show the curves.
3. Are there differences in final accuracy when you do the two? If not what is the value of showing how to evaluate? I would just entirely remove that as we have already proved the point via showing the loss curves. Unless there is a clear gap in the final performance, there is no need to confused reader with evaluation code.

Reply:
1. Ok I fixed this. The x- and y-axis of both runs are now uniform across the to pics to give a better perspective on how the losses related. I've also annotated the curves with their values to make it crystal clear what's happening.
2. I got rid of that whole part.
3. I got rid of the eval script.

Thanks for the feedback!